cognite/client/utils/_identifier.py

```diff
@@ -298,7 +298,7 @@ def load(cls, workflow_ids: Sequence[dict]) -> WorkflowVersionIdentifierSequence
                 [WorkflowVersionIdentifier(entry["version"], entry["workflowExternalId"]) for entry in workflow_ids],
                 is_singleton=False,
             )
-        raise TypeError(f"WorkflowIdentifier must be of type str or Sequence[str]. Found {type(workflow_ids)}")
+        raise TypeError(f"WorkflowIdentifier must be of type str or SequenceNotStr[str]. Found {type(workflow_ids)}")
```

Comment:
I'm not sure I agree we should put this in the error messages, might be very confusing.